### DISCUSSION TITLE
How to implement Nuxt 2's NuxtServerInit in a Nuxt 3/Pinia project?

### DISCUSSION THREAD

**Discussion by wmdev0808** [vuejs/pinia#2031] [Help and Questions]
I'm stuck with implementing Nuxt 2/NuxtServerInit in a Nuxt 3/Pinia project.

- Here is my repo for your reference: [project repo](https://github.com/wmdev0808/Vue/tree/main/challenge-002/02%20Handling%20Data/02-add-pinia)

  - The store is at `/stores/root.ts`
  - The plugin is at `/plugins/posts.ts`

- What I want to do:
  - When I run the app, the app loads the posts data, but I'd like to run the plugin only on server-side, so our Pinia store is loaded with default posts data in client-side.
  
- What the problem is:
  - To run the plugin only on server-side, I renamed the plugin to `plugins/posts.server.ts` and run the app again
    -  The plugin code only ran on the server-side
    - But the client-side didn't populate the data into the corresponding Pinia store
    
I hope you Pinia experts help me to fix this issue.
It would be better if you guys can make the PR for your fix into my repo.

Thank you in advance

**Comment by ahmadmobaraki** [comment]
You can do it with nuxt3 plugin, here is an example:
https://github.com/vuejs/pinia/discussions/947#discussioncomment-1943854
